src/hooks/useGridDimensions.ts

```diff
@@ -1,46 +1,63 @@
-import { useLayoutEffect, useRef, useState } from 'react';
-import { flushSync } from 'react-dom';
+import { useCallback, useRef, useState, useSyncExternalStore } from 'react';
 
-export function useGridDimensions() {
-  const gridRef = useRef<HTMLDivElement>(null);
-  const [inlineSize, setInlineSize] = useState(1);
-  const [blockSize, setBlockSize] = useState(1);
-  const [horizontalScrollbarHeight, setHorizontalScrollbarHeight] = useState(0);
+type Dimensions = [inlineSize: number, blockSize: number, horizontalScrollbarHeight: number];
 
-  useLayoutEffect(() => {
-    const { ResizeObserver } = window;
+const defaultDimensions: Dimensions = [1, 1, 0];
```

Comment:
Wonder if we should change the default to a higher value and get rid of initial width/height logic in the `subscribe` function